packages/react-storage/src/components/StorageBrowser/Views/LocationsView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,156 @@
+
+export type SortDirection = 'ascending' | 'descending' | 'none';
+
+export type SortState = {
+  selection: string;
+  direction: SortDirection;
+};
+
+const getCompareFn = (selection: string) => {
+  switch (selection) {
+    case 'scope':
+    case 'type':
+    case 'permission':
+      return compareStrings;
+  }
+};
+
+const getColumnItem = ({
+  entry,
+  selection,
+  direction,
+  onTableHeaderClick,
+}: {
+  entry: [string, string];
+  selection: string;
+  direction: SortDirection;
+  onTableHeaderClick: (location: string) => void;
+}) => {
+  const [key, value] = entry;
+
+  return {
+    children: (
+      <ButtonElement variant="sort" className={TABLE_HEADER_BUTTON_CLASS_NAME}>
+        {capitalize(value)}
+        <IconElement
+          variant={
+            selection === key && direction !== 'none'
+              ? `sort-${direction}`
+              : 'sort-indeterminate'
+          }
+        />
+      </ButtonElement>
+    ),
+    key: `th=${key}`,
+    className: `${TABLE_HEADER_CLASS_NAME} ${TABLE_HEADER_CLASS_NAME}--${key}`,
+    onClick: () => onTableHeaderClick(key),
+    'aria-sort': selection === key ? direction : 'none',
+  };
+};
+
+const displayColumns: Record<string, string>[] = [
+  { scope: 'name' },
+  { type: 'type' },
+  { permission: 'permission' },
+];
+
+const getLocationsData = ({
+  data,
+  onLocationClick,
+  onTableHeaderClick,
+  sortState,
+}: {
+  data: LocationAccess[];
+  onLocationClick: (location: LocationAccess) => void;
+  onTableHeaderClick: (location: string) => void;
+  sortState: SortState;
+}) => {
+  const { selection, direction } = sortState;
+
+  const columns = displayColumns.flatMap((column) =>
+    Object.entries(column).map((entry) =>
+      getColumnItem({ entry, selection, direction, onTableHeaderClick })
+    )
+  );
+
+  const compareFn = getCompareFn(selection);
+
+  if (compareFn) {
+    const castSelection = selection as keyof LocationAccess;
+
+    if (direction === 'ascending') {
+      data.sort((a, b) => compareFn(a[castSelection], b[castSelection]));
+    } else {
+      data.sort((a, b) => compareFn(b[castSelection], a[castSelection]));
+    }
+  }
+
+  const rows = data.map((location, index) => [
+    {
+      key: `td-name-${index}`,
+      children: (
+        <ButtonElement
+          className={TABLE_DATA_BUTTON_CLASS}
+          onClick={() => onLocationClick(location)}
+          variant="table-data"
+        >
+          {location.scope}
+        </ButtonElement>
+      ),
+    },
+    { key: `td-type-${index}`, children: location.type },
+    { key: `td-permission-${index}`, children: location.permission },
+  ]);
+
+  return { columns, rows };
+};
+
+export function DataTableControl(): React.JSX.Element {
+  const [{ data }] = useLocationsData();
+
+  const [, handleUpdateState] = useControl({ type: 'NAVIGATE' });
+
+  const [sortState, setSortState] = React.useState<SortState>({
+    selection: 'scope',
```

Comment:
Can we please stop using `scope` for reference values?

Reply:
I think I chose to use `scope` as the selection because that's how I would access it inside the location object when doing the compare. `name` isn't technically a property on the `LocationAccess` object is it?

Reply:
No it's not. For context here, exposing `scope` increases the need for conditional logic to be applied to convert it to `name`